src/main/java/stelnet/commodity/market/MarketRepository.java

```diff
@@ -0,0 +1,46 @@
+package stelnet.commodity.market;
+
+import java.util.HashMap;
+import java.util.List;
+
+import stelnet.commodity.CommodityTab;
+
+public class MarketRepository {
+
+    HashMap<String, List<MarketApiWrapper>> buyMarkets = new HashMap<>();
+    HashMap<String, List<MarketApiWrapper>> sellMarkets = new HashMap<>();
+    private static MarketRepository instance;
+
+    public MarketRepository(String commodityId) {
+        List<MarketApiWrapper> buyMarket = new BuyMarketFactory(commodityId).createMarkets();
+        List<MarketApiWrapper> sellMarket = new SellMarketFactory(commodityId).createMarkets();
+
+        buyMarkets.put(commodityId, buyMarket);
+        sellMarkets.put(commodityId, sellMarket);
+        instance = this;
+    }
+
+    public static MarketRepository getInstance(String commodityId) {
```

Comment:
So instead I store the reference in a board, which itself is rooted in IntelManagerAPI. Then a public getter for it.